File: Client_Root/Client/Assets/Scripts/Network/Messages/ToRoom/GameInputRotationToR.cs
using UnityEngine;
using FlatBuffers;

public class GameInputRotationToR : IMessage
{
    public const ushort MESSAGE_ID = MessageID.GameInputRotationToR_ID;

    public int m_nPlayerIndex;
    public Vector3 m_Rotation;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        FBS.GameInputRotationToR.StartGameInputRotationToR(m_Builder);
        FBS.GameInputRotationToR.AddPlayerIndex(m_Builder, m_nPlayerIndex);
        FBS.GameInputRotationToR.AddRotation(m_Builder, FBS.Data.Vector3.CreateVector3(m_Builder, m_Rotation.x, m_Rotation.y, m_Rotation.z));
        var data = FBS.GameInputRotationToR.EndGameInputRotationToR(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.GameInputRotationToR.GetRootAsGameInputRotationToR(buf);

        m_nPlayerIndex = data.PlayerIndex;
        m_Rotation = new Vector3(data.Rotation.X, data.Rotation.Y, data.Rotation.Z);

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ToRoom/GameInputRotationToR.cpp
#include "stdafx.h"
#include "GameInputRotationToR.h"
#include "../../../FBSFiles/GameInputRotationToR_generated.h"

GameInputRotationToR::GameInputRotationToR()
{
}

GameInputRotationToR::~GameInputRotationToR()
{
}

unsigned short GameInputRotationToR::GetID()
{
	return MESSAGE_ID;
}

IMessage* GameInputRotationToR::Clone()
{
	GameInputRotationToR* pClone = new GameInputRotationToR();

	pClone->m_nPlayerIndex = m_nPlayerIndex;
	pClone->m_Rotation = m_Rotation;

	return pClone;
}

const char* GameInputRotationToR::Serialize(int* pLength)
{
	FBS::Data::Vector3 rotation(m_Rotation.x(), m_Rotation.y(), m_Rotation.z());

	FBS::GameInputRotationToRBuilder data_builder(m_Builder);
	data_builder.add_PlayerIndex(m_nPlayerIndex);
	data_builder.add_Rotation(&rotation);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool GameInputRotationToR::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::GameInputRotationToR>((const void*)pChar);

	m_nPlayerIndex = data->PlayerIndex();
	m_Rotation.setValue(data->Rotation()->x(), data->Rotation()->y(), data->Rotation()->z());

	return true;
}